src/tpm2-tss-engine.c

```diff
@@ -165,16 +166,50 @@ loadkey(ENGINE *e, const char *key_id, UI_METHOD *ui, void *cb_data)
     EVP_PKEY *pkey = NULL;
 
 	DBG("Loading private key %s\n", key_id);
-
-    if (!tpm2tss_tpm2data_read(key_id, &tpm2Data)) {
+    tpm2Data = OPENSSL_malloc(sizeof(*tpm2Data));
+    if (tpm2Data == NULL) {
+      ERR(tpm2tss_tpm2data_read, ERR_R_MALLOC_FAILURE);
+      goto error;
+    }
+    if (strncmp(key_id, "0x", 2) == 0) {
```

Comment:
Please let's standardise a URI format for this, perhaps based on https://tools.ietf.org/html/draft-mavrogiannopoulos-tpmuri-01

Reply:
That thing is outdated for TPM2...
No uuids anymore and "storage" has different means of system vs user...
I'd wanna go with 0x or 0x81 for the moment, until we have figured out a tpm2uri scheme or adapted tpmuri...